pkg/client/didexchange/client.go

```diff
@@ -36,15 +40,24 @@ func New(ctx provider) (*Client, error) {
 	if !ok {
 		return nil, errors.New("cast service to DIDExchange Service failed")
 	}
-	return &Client{didexchangeSvc: didexchangeSvc}, nil
+	return &Client{didexchangeSvc: didexchangeSvc, wallet: ctx.Wallet()}, nil
 }
 
 // CreateInvitation create invitation
 func (c *Client) CreateInvitation() (*InvitationRequest, error) {
+	// TODO remove nil check after provide default implementation for wallet
```

Comment:
This TODO seems related?

Reply:
wallet can be nil because there is no default implementation for wallet, after adding the default implementation I will remove the nil check